theorems/cohomology/CupProduct/Definition.agda

```diff
@@ -0,0 +1,304 @@
+{-# OPTIONS --without-K --rewriting #-}
+
+open import HoTT
+open import cohomology.CupProduct.OnEM.InAllDegrees
+open import cohomology.CupProduct.OnEM.CommutativityInAllDegrees
+open import cohomology.EMModel
+open import cohomology.Theory
+open import groups.ToOmega
+open import homotopy.EilenbergMacLane
+open import homotopy.EilenbergMacLaneFunctor
+open import homotopy.Freudenthal
+open import homotopy.SuspensionLoopSpaceInverse
+
+module cohomology.CupProduct.Definition {i} (X : Ptd i) where
+
+private
+  module M {k} (A : AbGroup k) = CohomologyTheory (EM-Cohomology A)
+  open M
+
+smin-map : ∀ {j k} {Y : Ptd j} {Z : Ptd k}
+  → X ⊙→ Y
+  → X ⊙→ Z
+  → X ⊙→ Y ⊙∧ Z
+smin-map f g = (λ x → smin (fst f x) (fst g x)) , ap2 smin (snd f) (snd g)
+
+smin-map-⊙∧-swap : ∀ {j k} (Y : Ptd j) (Z : Ptd k)
+  (f : X ⊙→ Y)
+  (g : X ⊙→ Z)
+  → ⊙∧-swap Z Y ⊙∘ smin-map g f == smin-map f g
+smin-map-⊙∧-swap Y Z f g =
+  ⊙λ=' (λ x → idp) $
+  ap (∧-swap Z Y) (ap2 smin (snd g) (snd f)) ∙ idp
+    =⟨ ∙-unit-r (ap (∧-swap Z Y) (ap2 smin (snd g) (snd f))) ⟩
+  ap (∧-swap Z Y) (ap2 smin (snd g) (snd f))
+    =⟨ ap-ap2 (∧-swap Z Y) smin (snd g) (snd f) ⟩
+  ap2 (λ x y → smin y x) (snd g) (snd f)
+    =⟨ =ₛ-out $ ap2-out (λ x y → smin y x) (snd g) (snd f) ⟩
+  ap (smin (fst f (pt X))) (snd g) ∙ ap (λ y → smin y (pt Z)) (snd f)
+    =⟨ ! $ =ₛ-out $ ap2-out' smin (snd f) (snd g) ⟩
+  ap2 smin (snd f) (snd g) =∎
+
+module _ (G : AbGroup i) (H : AbGroup i) where
+
+  private
+    module G⊗H = TensorProduct G H
+    module H⊗G = TensorProduct H G
+  open EMExplicit
+
+  ⊙Ω∧-cp-seq : ∀ (m n : ℕ) → (⊙Ω (⊙EM G (S m)) ⊙∧ ⊙Ω (⊙EM H (S n))) ⊙–→ ⊙Ω (⊙EM G⊗H.abgroup (S (m + n)))
+  ⊙Ω∧-cp-seq m n =
+    ⊙<– (spectrum G⊗H.abgroup (m + n)) ◃⊙∘
+    ⊙∧-cp G H m n ◃⊙∘
+    ⊙∧-fmap (⊙–> (spectrum G m)) (⊙–> (spectrum H n)) ◃⊙idf
+
+  ⊙Ω∧-cp : ∀ (m n : ℕ) → ⊙Ω (⊙EM G (S m)) ⊙∧ ⊙Ω (⊙EM H (S n)) ⊙→ ⊙Ω (⊙EM G⊗H.abgroup (S (m + n)))
+  ⊙Ω∧-cp m n = ⊙compose (⊙Ω∧-cp-seq m n)
+
+
+  _∪_  : ∀ {m n : ℕ} → CEl G (pos m) X → CEl H (pos n) X → CEl G⊗H.abgroup (pos (m + n)) X
+  _∪_ {m} {n} s t = Trunc-rec (λ s' → Trunc-rec (λ t' → [ ⊙Ω∧-cp m n ⊙∘ smin-map s' t' ]) t) s
+
+  -- ∪-hom : ∀ {m n : ℕ} → TensorProduct.grp (C-abgroup G (pos m) X) (C-abgroup H (pos n) X) →ᴳ C G⊗H.abgroup (pos (m + n)) X
+  -- ∪-hom = {!!}
```

Comment:
Is this something you are planning to do in the future?

Reply:
Currently I'm not planning to do this. I have already deleted this comment.